### ISSUE TITLE
Issue with AnyStr within a Union

### ISSUE THREAD

**Issue by pgjones** [python/mypy#14466]
The following does not raise any issues, yet I think it should: 

```python
from typing import AnyStr, Union

RV = Union[AnyStr]

b: RV = {}, {}, 429 
```

```
$ mypy --version
mypy 0.991 (compiled: yes)
$ python --version
Python 3.11.1
```

For context (aside), the above is a simplified version of:

```python
from quart import ResponseReturnValue

response: ResponseReturnValue = {}, {}, 429 
```

Which should error and let me know that it should have been `{}, 429, {}`. See also, [quart typing](https://github.com/pallets/quart/blob/main/src/quart/typing.py#L51)

**Comment by pgjones** [comment]
Actually I think this is a misuse of AnyStr, closing
